projects/tutorials/object_nav_ithor_ppo_baseline.py

```diff
@@ -0,0 +1,381 @@
+            depth_resnet_preprocessor_uuid="depth_resnet" if has_depth else None,
+            hidden_size=512,
+            goal_dims=32,
+        )
+
+    @classmethod
+    def make_sampler_fn(cls, **kwargs) -> TaskSampler:
+        return ObjectNavDatasetTaskSampler(**kwargs)
+
+    @staticmethod
+    def _partition_inds(n: int, num_parts: int):
+        return np.round(np.linspace(0, n, num_parts + 1, endpoint=True)).astype(
+            np.int32
+        )
+
+    def _get_sampler_args_for_scene_split(
+        self,
+        scenes: List[str],
+        process_ind: int,
+        total_processes: int,
+        seeds: Optional[List[int]] = None,
+        deterministic_cudnn: bool = False,
+    ) -> Dict[str, Any]:
+        if total_processes > len(scenes):  # oversample some scenes -> bias
+            if total_processes % len(scenes) != 0:
+                print(
+                    "Warning: oversampling some of the scenes to feed all processes."
+                    " You can avoid this by setting a number of workers divisible by the number of scenes"
+                )
+            scenes = scenes * int(ceil(total_processes / len(scenes)))
+            scenes = scenes[: total_processes * (len(scenes) // total_processes)]
+        else:
+            if len(scenes) % total_processes != 0:
+                print(
+                    "Warning: oversampling some of the scenes to feed all processes."
+                    " You can avoid this by setting a number of workers divisor of the number of scenes"
+                )
+        inds = self._partition_inds(len(scenes), total_processes)
+
+        return {
+            "scenes": scenes[inds[process_ind] : inds[process_ind + 1]],
+            "object_types": self.OBJECT_TYPES,
+            "env_args": self.ENV_ARGS,
+            "max_steps": self.MAX_STEPS,
+            "sensors": self.SENSORS,
+            "action_space": gym.spaces.Discrete(
+                len(ObjectNaviThorGridTask.class_action_names())
+            ),
+            "seed": seeds[process_ind] if seeds is not None else None,
+            "deterministic_cudnn": deterministic_cudnn,
+        }
+
+    def train_task_sampler_args(
+        self,
+        process_ind: int,
+        total_processes: int,
+        devices: Optional[List[int]] = None,
+        seeds: Optional[List[int]] = None,
+        deterministic_cudnn: bool = False,
+    ) -> Dict[str, Any]:
+        res = self._get_sampler_args_for_scene_split(
+            self.TRAIN_SCENES,
+            process_ind,
+            total_processes,
+            seeds=seeds,
+            deterministic_cudnn=deterministic_cudnn,
+        )
+        res["scene_directory"] = os.path.join(
+            os.getcwd(), "datasets/ithor-objectnav/train/episodes"
+        )
+        res["loop_dataset"] = True
+        res["scene_period"] = "manual"
+        res["env_args"] = {}
+        res["env_args"].update(self.ENV_ARGS)
+        res["env_args"]["x_display"] = (
```

Comment:
Not wrong but a more resilient way to find the `x_displays` is to do what is done [here](https://github.com/allenai/allenact/blob/c9485aa39ec3eec89b74984a089119bf407c84d1/projects/pointnav_baselines/experiments/pointnav_thor_base.py#L158).